From f088a6f63d0eddbd6db4eefbe55f6b9921ef913d Mon Sep 17 00:00:00 2001
From: Julian Bouzas <julian.bouzas@collabora.com>
Date: Thu, 8 Jan 2026 13:47:21 -0500
Subject: [PATCH] autoswitch-bluetooth-profile: Fix attempt to index a number
 value error

The getLinkedBluetoothLoopbackSourceNodeForStream() function expects stream to be
and object and not a stream ID.
---
 src/scripts/device/autoswitch-bluetooth-profile.lua | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/scripts/device/autoswitch-bluetooth-profile.lua b/src/scripts/device/autoswitch-bluetooth-profile.lua
index 07f8db39cfca..ee41c32e35a1 100644
--- a/src/scripts/device/autoswitch-bluetooth-profile.lua
+++ b/src/scripts/device/autoswitch-bluetooth-profile.lua
@@ -314,8 +314,7 @@ function getLinkedBluetoothLoopbackSourceNodeForStream (stream, node_om, link_om
           Constraint { "bluez5.loopback", "!", "true", type = "pw" },
           Constraint { "node.link-group", "=", filter_link_group, type = "pw" }
         } do
-        local filter_stream_id = filter_stream_node["bound-id"]
-        local bt_node = getLinkedBluetoothLoopbackSourceNodeForStream (filter_stream_id, node_om, link_om, visited_link_groups)
+        local bt_node = getLinkedBluetoothLoopbackSourceNodeForStream (filter_stream_node, node_om, link_om, visited_link_groups)
         if bt_node ~= nil then
           return bt_node
         end
-- 
2.52.0

